### ISSUE TITLE
Where are the published CI build logs, specifically for fsharpqa?

### ISSUE THREAD

**Issue by abelbraaksma** [dotnet/fsharp#9869]
Since it's _very_ hard to run tests locally (I've figured it out and will write up the results soon, but also since `HostedCompilerService.exe` keeps running after closing VS, keeping files locked), and since running them takes a lot of time, it would be great if we could download the logs with the errors from the CI results.

The logs are published (see, for instance, this: https://dev.azure.com/dnceng/public/_build/results?buildId=755330&view=logs&j=f63a89e5-f92e-5427-3260-3ba3a585220c&t=a75259a2-1178-52d5-2540-928aa6fdbf8d)

But when you go to a CI node, click "artifacts" and then click the necessary download, nothing happens: it is not downloaded. See https://dev.azure.com/dnceng/public/_build/results?buildId=755330&view=artifacts&type=publishedArtifacts

The regular log _does not contain the error output_, it only shows "failed" for fsharpqa tests, which, in my case, leads to "guessing what the test expects"... (add confused smiley here).

https://github.com/dotnet/fsharp/pull/9779

**Comment by cartermp** [comment]
You can download logs from the top-left of the CI portal:

<img width="254" alt="image" src="https://user-images.githubusercontent.com/6309070/89307267-b9596900-d625-11ea-86e9-a1657ce2a368.png">

However there are no diagnostic logs published for the fsharpqa tests as far as I can tell. This is one of the reasons why they're so awful :)

Perhaps there's a way to publish them but since it's a bespoke framework instead of a standardized one like NUnit or xUnit, it's unlikely that most systems know how to properly display logs unless it's just some plaintext.

**Comment by abelbraaksma** [comment]
> However there are no diagnostic logs published for the fsharpqa tests as far as I can tell. This is one of the reasons why they're so awful :)

Indeed ;). I thought this was the key:

![image](https://user-images.githubusercontent.com/16015770/89310876-93f44b80-d675-11ea-8077-21a17adff098.png)

And then this:

![image](https://user-images.githubusercontent.com/16015770/89310997-b8502800-d675-11ea-9ee2-88a2079d95ca.png)

But nothing gets downloaded when I click that.

Locally, an error log is created, though.

**Comment by abelbraaksma** [comment]
In the `build.ps1` it has this:

```ps
        $resultsLog = "test-net40-fsharpqa-results.log"
        $errorLog = "test-net40-fsharpqa-errors.log"
        $failLog = "test-net40-fsharpqa-errors"
```

**Comment by abelbraaksma** [comment]
Looks like the following command (and actions) gets this running locally:

* Close all VS instances
* Close, from Task Mgr, or kill, `HostedCompilerService.exe` and/or `VBSCompiler.exe`. You may also kill any hanging watson processes
* Run this from the root of the repo: `build -testFSharpQA -ci -bl -c Release`
   * Note that without `-c Release`, you'll get a bunch of more errors and assertion-popups
   * Without closing processes, it may not compile, or errors/crashes and popups half way
   * Without `-ci -bl` this the tests won't run properly at all
* It may appear as if the process hangs after `HexCharEncode.fs`, but the next test just takes 4-6 min to complete

After this, the log file `[basedir]artifacts\TestResults\Release` contains the log information you need.

**Comment by dsyme** [comment]
I've always struggled to find the logs for `fsharpqa` too

**Comment by abelbraaksma** [comment]
@dsyme I don't know how long it may take to get these converted to the new test system, but if it is still several months away, a simple tweak to CI would be to just upload these auxiliary artifacts, even though they aren't binary. They are there, but cannot be accessed right now.

**Comment by abelbraaksma** [comment]
@dsyme, @cartermp, I suddenly found out where these logs are (after I found out that the `azure-pipelines.fsy` were correct). It's one of those things that _once you know it, it's easy to find_. I made an edited gif-video tutorial, we can add it to the contributors instruction:

![484e5c1e-2f63-41c7-af12-146f5e49ab89](https://user-images.githubusercontent.com/16015770/91312892-065bd580-e7b5-11ea-95db-ed9695389906.gif)

@pblasucci, you struggled with this in your most recent PR as well. Now you know where to access them, even if you cannot build them locally.
